autotests/testolmutility.cpp

```diff
@@ -0,0 +1,129 @@
+// SPDX-FileCopyrightText: 2021 Carl Schwan <carlschwan@kde.org>
+//
+// SPDX-License-Identifier: LGPL-2.1-or-later
+
+#include "testolmutility.h"
+#include "e2ee/qolmaccount.h"
+#include "e2ee/qolmutility.h"
+
+using namespace Quotient;
+
+void TestOlmUtility::canonicalJSON()
+{
+    // Examples taken from
+    // https://matrix.org/docs/spec/appendices.html#canonical-json
+    auto data = QJsonDocument::fromJson(QByteArrayLiteral(R"({
+    "auth": {
+      "success": true,
+      "mxid": "@john.doe:example.com",
+      "profile": {
+        "display_name": "John Doe",
+        "three_pids": [{
+          "medium": "email",
+          "address": "john.doe@example.org"
+        }, {
+          "medium": "msisdn",
+          "address": "123456789"
+        }]
+      }}})"));
+
+    QCOMPARE(data.toJson(QJsonDocument::Compact),
+      "{\"auth\":{\"mxid\":\"@john.doe:example.com\",\"profile\":{\"display_name\":\"John "
+      "Doe\",\"three_pids\":[{\"address\":\"john.doe@example.org\",\"medium\":\"email\"},{"
+      "\"address\":\"123456789\",\"medium\":\"msisdn\"}]},\"success\":true}}");
+
+    auto data0 = QJsonDocument::fromJson(QByteArrayLiteral(R"({"b":"2","a":"1"})"));
+    QCOMPARE(data0.toJson(QJsonDocument::Compact), "{\"a\":\"1\",\"b\":\"2\"}");
+
+    auto data1 = QJsonDocument::fromJson(QByteArrayLiteral(R"({ "本": 2, "日": 1 })"));
+    QCOMPARE(data1.toJson(QJsonDocument::Compact), "{\"日\":1,\"本\":2}");
+
+    auto data2 = QJsonDocument::fromJson(QByteArrayLiteral(R"({"a": "\u65E5"})"));
+    QCOMPARE(data2.toJson(QJsonDocument::Compact), "{\"a\":\"日\"}");
+
+    auto data3 = QJsonDocument::fromJson(QByteArrayLiteral(R"({ "a": null })"));
+    QCOMPARE(data3.toJson(QJsonDocument::Compact), "{\"a\":null}");
+}
+
+void TestOlmUtility::verifySignedOneTimeKey()
+{
+    auto aliceOlm = std::make_shared<QOlmAccount>("alice:matrix.org", "aliceDevice");
+    aliceOlm->createNewAccount();
+    aliceOlm->generateOneTimeKeys(1);
+    auto keys = aliceOlm->oneTimeKeys();
```

Comment:
Why `make_shared` if it's entirely local? Also, is `@` missing in MXID?
```suggestion
    QOlmAccount aliceOlm { "@alice:matrix.org", "aliceDevice" };
    aliceOlm.createNewAccount();
    aliceOlm.generateOneTimeKeys(1);
    auto keys = aliceOlm.oneTimeKeys();
```